Hiv cs tx continuity

DennisGibz opened the pull request.

nobert-mumo commented on Scripts/REPORTING/HIVCaseSurveillance/load_cs_TxContinuity.sql:
```diff
@@ -8,19 +8,20 @@ BEGIN
 			,Patient.Gender
 			,Patient.DOB
-			,[AgeGroup].DATIMAgeGroup
+			,FactARTHistory.AgeGroup
```
@DennisGibz not sure, I thought since this is a Fact we should be using an AgeGroupKey?

nobert-mumo replied:
@DennisGibz please recheck on this.

nobert-mumo reviewed (changes requested):
@DennisGibz not sure, I thought since this is a Fact we should be using an AgeGroupKey?

DennisGibz commented:
@nobert-mumo yes but DimAgegroup was bringing about discrepancy in numbers. We have changed the report to be a linelist.

nobert-mumo commented:
> @nobert-mumo yes but DimAgegroup was bringing about discrepancy in numbers. We have changed the report to be a linelist.

@DennisGibz not sure I follow, do you mean we changed the fact to a report ?

Marymary-dev commented on Scripts/REPORTING/HIVCaseSurveillance/load_cs_TxContinuity.sql:
```diff
@@ -23,7 +23,7 @@ BEGIN
 			,YEAR(TRY_CAST(DateConfirmedHIVPositiveKey AS DATETIME2)) As CohortYear
 			,TRY_CAST(DateConfirmedHIVPositiveKey AS DATETIME2) As CohortYearMonth
```
Ensure the CohortYearMonth is set to EndofMonth for ease of filtering

DennisGibz replied:
@Marymary-dev this has since been addressed.

Marymary-dev reviewed (commented):
Ensure the CohortYearMonth is set to EndofMonth for ease of filtering

DennisGibz commented:
> > @nobert-mumo yes but DimAgegroup was bringing about discrepancy in numbers. We have changed the report to be a linelist.
> 
> @DennisGibz not sure I follow, do you mean we changed the fact to a report ?

@nobert-mumo  this was never a fact. We have factARTHistory as the source fact.

nobert-mumo commented:
> > > @nobert-mumo yes but DimAgegroup was bringing about discrepancy in numbers. We have changed the report to be a linelist.
> > 
> > 
> > @DennisGibz not sure I follow, do you mean we changed the fact to a report ?
> 
> @nobert-mumo this was never a fact. We have factARTHistory as the source fact.

@DennisGibz I thought [NDWH].[dbo].[FactARTHistory] is a fact ? then we should be having an `AgeGroupKey` that we join to `DimAgeGroup` when building the report.

DennisGibz commented:
> > > > @nobert-mumo yes but DimAgegroup was bringing about discrepancy in numbers. We have changed the report to be a linelist.
> > > 
> > > 
> > > @DennisGibz not sure I follow, do you mean we changed the fact to a report ?
> > 
> > 
> > @nobert-mumo this was never a fact. We have factARTHistory as the source fact.
> 
> @DennisGibz I thought [NDWH].[dbo].[FactARTHistory] is a fact ? then we should be having an `AgeGroupKey` that we join to `DimAgeGroup` when building the report.

Yeah. We can have the dim attached to the fact then consume the column downstream. I'll look at that.

nobert-mumo commented:
@DennisGibz , the table `FactARTHistory` seems to be a bit off from being a fact table? I think we need to design not to have the table denormalized but only contain the foreign dim keys

DennisGibz commented:
The PR has since been modified @nobert-mumo , @Marymary-dev , @cbrianbet

nobert-mumo commented on Scripts/REPORTING/HIVCaseSurveillance/load_cs_TxContinuity.sql:
```diff
@@ -8,38 +8,37 @@ BEGIN
-			,TRY_CAST(DateConfirmedHIVPositiveKey AS DATETIME2) As CohortYearMonth
-			,cast(asofdatekey as datetime2) As OutcomeYearMonth
+			,EOMONTH(TRY_CAST(DateConfirmedHIVPositiveKey AS DATETIME2)) As CohortYearMonth
 			,COUNT(1) AS NoOfClients
```
@DennisGibz, do we need this column here since this is a patient level dataset ?

nobert-mumo commented on Scripts/REPORTING/HIVCaseSurveillance/load_cs_TxContinuity.sql:
```diff
@@ -39,21 +36,9 @@ BEGIN
-				,asofdatekey
-				,DateConfirmedHIVPositiveKey
-				,[AgeGroup].DATIMAgeGroup
+		WHERE Facility.MFLCode IS NOT NULL AND FactARTHistory.ARTOutcomeKey in (2,3,6,8)
```
@DennisGibz maybe we can replace the ARTOutcome with ARTOutcomeDescription so that it 'easily observable'? I know the DimKey don't change but the description would be better IMO.

nobert-mumo commented:
@DennisGibz the model seems to contain duplicates for Patient, AsOfDate combinations, can you re-check?

`select 
    PatientPKHash,
    MFLCode,
    asofdate,
    count(*) as count
from [HIVCaseSurveillance].[dbo].[CsTxContinuity]
group by 
    PatientPKHash,
    MFLCode,
    asofdate
having count(*) > 1`
![image](https://github.com/user-attachments/assets/fd4468d9-d04d-4494-9b8c-651c5f2aea4c)

nobert-mumo reviewed (approved):
LGTM

DennisGibz commented:
@Marymary-dev  uko na maoni?
